### ISSUE TITLE
Editing a commit content does not update the existing commit

### ISSUE THREAD

**Issue by SicroAtGit** [sublimehq/sublime_merge#1533]
**Version info**

 - OS: Manjaro Linux
 - Build: 2074

**Description**

When I click on "Edit Commit - Edit Commit Contents" in the context menu of a commit, make the change to the file and stage the changed file, in Sublime Merge I only have the option to replace the existing commit with a new one or amend a previous commit. There is no possibility to only continue the rebase process so that the existing commit is updated.

When I click on `Commit`, the existing commit is replaced by a new commit instead of updating the existing commit, which results in the original author being replaced by me instead of naming me as the committer.

When I click on `Continue rebase`, Sublime Merge shows me the error message:

> error:
You have uncommitted changes in your working directory. Please commit them first and then run `git rebase --continue` again.

What is interesting is when I initiate the rebase process not in Sublime Merge, but with Git directly, then it works:

1. `git rebase -i commit^`
2. Change `pick` to `edit` for the commit to be edited
3. Do the changes
4. In Sublime Merge: Stage the changed file
5. In Sublime Merge: Click on `Continue rebase`

**Expected behavior**

It should be possible to continue rebasing without creating a new commit or amending the previous one so that the existing commit is updated.

With Git it also works:

1. `git rebase -i commit^`
2. Change `pick` to `edit` for the commit to be edited
3. Do the changes
4. `git add changed_file`
5. `git rebase --continue`

**Comment by dpjohnst** [comment]
Hi @SicroAtGit,

Thanks for reaching out!

The way `Edit Commit Contents` currently works is that it will undo all of the commits up to and including the selected commit.
All changes from this commit are then placed into the staged changes area, allowing you to directly select and commit changes (rather than amending and reverting lines from the previous commit).

A current known limitation of this is that when you re-commit, the author information is lost. A solution for this is currently being addressed. I'll keep you updated as things progress.

Kind regards,
\- Dylan from Sublime HQ

**Comment by srbs** [comment]
duplicates: #728

**Comment by srbs** [comment]
workaround posted here: https://github.com/sublimehq/sublime_merge/issues/728#issuecomment-1208858702

**Comment by SicroAtGit** [comment]
@dpjohnst 

Yes, the current approach is also useful, e.g. to split a commit into several new commits. So it may make sense to allow both approaches.

---

@srbs

Thanks, the workaround is very helpful.

Because such an issue already exists, mine can be closed. I have activated notifications for the other open issue as well.
